src/components/common/Text.jsx

```diff
@@ -63,51 +63,41 @@ const DropdownContent = styled.div`
   ${props => props.dropdownMargin && css`
     margin: ${props.dropdownMargin};
   `}
-`;
-
-class Text extends Component {
-  constructor(props) {
-    super(props);
-
-    this.toogleDropdown = this.toogleDropdown.bind(this);
+  `;
 
-    this.state = {
-      dropdownOpen: false,
-    };
-  }
+function Text(props) {
+  const [dropdownOpen, setDropdownOpen] = useState(false);
 
-  toogleDropdown() {
-    const currentState = this.state.dropdownOpen;
-    this.setState({
-      dropdownOpen: !currentState,
-    });
+  const toggleDropdown = () => {
+    setDropdownOpen(!dropdownOpen);
   }
 
-  render() {
-    let Selected;
-    switch (this.props.type) {
+  const getSelected = () => {
+    switch (props.type) {
       case 'paragraph':
-        Selected = Paragraph;
-        break;
+        return Paragraph;
       case 'header':
-        Selected = Header;
-        break;
+        return Header;
       default:
-        Selected = Span;
-        break;
+        return Span;
     }
```

Comment:
Definitely don't use arrow functions inside an inner scope , move it to the outer scope & check this https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/consistent-function-scoping.md